pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -266,12 +276,40 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
 					if errors.Is(err, context.Canceled) {
 						return nil
 					}
+					localCEP, err = ciliumClient.CiliumEndpoints(namespace).Patch(
```

Comment:
Non-blocking review, but can we detect the error returned from kube-apiserver to understand if we can perform a JSON patch directly to the resource (instead of the `status` sub-resource)? My main concern is that in a worst case scenario we will perform yet another operation to kube-apiserver trying to update the CEP and kube-apiserver will returned the same error as returned on line 255.

Reply:
Yea, I thought about this too and decided it was less of a concern for us since this is only supposed to be an intermediate version with no known large-scale clusters in use. But I think you are right, for upstream this might be more of an issue. I will check for the error and only try direct patch if needed.

Reply:
Done.